### PR TITLE
cmake: Adjust `test/README.md` instructions

### PR DESCRIPTION
**Pull request by paplorinc**
Addresses one readme from https://github.com/bitcoin/bitcoin/pull/30454#discussion_r1711661191.

Let me know if this is the direction you want to go or not, before I attempt to migrate the rest.

Feel free to push commits on top instead of explaining, if that's simpler.

### CODE REVIEW

File: src/test/README.md [hunk 1]
```diff
@@ -12,22 +12,30 @@ that runs all of the unit tests. The main source file for the test library is fo
...
 ### Compiling/running unit tests
 
-Unit tests will be automatically compiled if dependencies were met in `./configure`
-and tests weren't explicitly disabled.
+Unit tests will be automatically compiled if the dependencies are met during the CMake configuration process
+and the tests haven't been explicitly disabled.
 
-After configuring, they can be run with `make check`, which includes unit tests from
-subtrees, or `make && make -C src check-unit` for just the unit tests.
+To configure the project, build it in the 'build' directory, and then run all unit tests using 8 threads,
```
Comment (hebasto) [inline src/test/README.md:18]: Not sure if adding parallelism options is related to the migration to CMake. In any case, explicitly mentioning a fixed number of threads gives the impression that this number is a hard limit and that the option itself is obligatory.

If you insist on mentioning the `-j` option, you could follow the pattern from the build notes.

**Review by hebasto — COMMENTED** [review]
@paplorinc 

Thank you for your contribution!

There are a few more instances `src/test/test_bitcoin` that should be prefixed with the build directory.

File: src/test/README.md [hunk 2]
```diff
@@ -12,22 +12,30 @@ that runs all of the unit tests. The main source file for the test library is fo
...
+  && cmake --build build -j8 \
+  && ctest --test-dir build -j8
+```
+
+To manually run a specific unit test (e.g., `bech32_testvectors_valid` in `bech32_tests.cpp`), use the following commands:
+```bash
+cmake -B build \
+  && cmake --build build -j8 --target src/test/test_bitcoin \
+  && build/src/test/test_bitcoin --run_test=bech32_tests/bech32_testvectors_valid
+```
```
[the remark this answers is not in the document]
Comment (paplorinc) [reply src/test/README.md]: Agree, moved it to the beginning of the `Running individual tests` section

File: src/test/README.md [hunk 3]
```diff
@@ -12,22 +12,30 @@ that runs all of the unit tests. The main source file for the test library is fo
 
 ### Compiling/running unit tests
 
-Unit tests will be automatically compiled if dependencies were met in `./configure`
-and tests weren't explicitly disabled.
+Unit tests will be automatically compiled if the dependencies are met during the CMake configuration process
```
[the remark this answers is not in the document]
Comment (paplorinc) [reply src/test/README.md]: Done

**Comment by paplorinc** [comment]
> There are a few more instances src/test/test_bitcoin that should be prefixed with the build directory.

Done

File: src/test/README.md (hunk 1 above)
Comment (paplorinc) [reply src/test/README.md]: Done

File: src/test/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (paplorinc) [reply src/test/README.md]: Changed to
> Unit tests will be automatically compiled if the dependencies are met during the generation of the Bitcoin Core build system

instead

**Review by hebasto — COMMENTED** [review]
d281ef341f6b56d8eefc8e080be00000fa6842db looks better!

However, I still believe that this documentation update PR should focus on changes related to the migration from Autotools to CMake, leaving other improvements for separate PRs.

**Comment by paplorinc** [comment]
Seems it would be faster for you to edit it in that case - feel free to push on top or close and cherry pick.

**Comment by hebasto** [comment]
> Seems it would be faster for you to edit it in that case - feel free to push on top or close and cherry pick.

Picked in https://github.com/hebasto/bitcoin/pull/328.

**Pull request closed without merging**